docs/custom-tests/05-bootstrap-and-test-phases.md

```diff
@@ -0,0 +1,132 @@
+import CustomTestsDiagram from '@site/src/img/custom-tests-diagram.png';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Bootstrap and Test Phases
+
+:::info
+The Custom Tests feature is coming soon.
+:::
+
+## Introduction
+
+The E2E Tests have three key parts: the **Bootstrap Phase**, the **Entrypoint**, and the **Test Phase**.
+
+This is the flow of execution:
+
+<img src={CustomTestsDiagram}/>
+
+## The Bootstrap Phase
+
+In this phase, set up everything your tests needs, and do the initial setup of your plugin.
+
+If you plugin have an onboarding wizard, this is where you need to configure it so that it doesn't show up anymore. Get rid of any data consent forms, or any other setup that needs to be done before the tests run.
+
+If your plugin integrates with an external service, here's where you can configure mock responses, enable a development mode.
+
+If your tests expects a specific theme to be active, you **install** them in this step so that they are available on your Test Phase, and you **activate it** in the beginning of your Test Phase eg:
+
+**bootstrap.sh**
+
+```bash
+# Install Storefront Theme so that we can activate it on our Test phase.
+wp theme install storefront
+```
+
+### `bootstrap.sh`
+
+This is a bash script that, if present, will be called. Here you can use `wp` (WP-CLI) and bash in general.
+
+### `bootstrap.php`
+
+This is a PHP file that, if present, will be called. WordPress is NOT loaded on the context of this file.
+
+### `must-use-plugin.php`
+
+This is a file that, if present, will be copied to the `wp-content/mu-plugins` directory and will run on ALL requests on the E2E test. WordPress is loaded on the context of this file.
+
+If you want to put logic in there that should run only once, make sure to set a flag to prevent it from running multiple times.
+
+### Where to Put the Bootstrap Files
+
+Place them in the bootstrap directory within your E2E test suite. Here’s how your directory structure should look:
+
+```
+bootstrap             (Optional)
+  bootstrap.php       (Optional)
+  bootstrap.sh        (Optional)
+  must-use-plugin.php (Optional)
```

Comment:
Should we note anything on the naming of this file? I believe it has to be this format, to get picked up, so it might be good to note that.